view_conversations/conv-viewer-webapi/src/db_types.ts

```diff
@@ -0,0 +1,193 @@
+/*
+Copyright 2018 Google Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+import * as spanner from '@google-cloud/spanner';
+import * as runtime_types from './runtime_types';
+
+var ab2str = require('arraybuffer-to-string')
+
+// Spanner Row Representations in JS.
+//
+// Note: Cloud Spanner interprets Node.js numbers as FLOAT64s, so
+// in the row format, they are treated as strings (spanner does
+// the conversion to INT64s when appropriate)
+// TODO (yiqingh): modify this part accordingly
+export interface OutputRow {
+  id: runtime_types.CommentId;
+  ancestor_id: runtime_types.CommentId;
+  authors: null | string[];
+  cleaned_content: null | string;
+  content: null | string;
+  conversation_id: runtime_types.ConversationId;
+  indentation: number;
+  page_id: number;
+  page_title: string;
+  parent_id: runtime_types.CommentId;
+  replyTo_id: runtime_types.CommentId;
+  rev_id: runtime_types.RevisionId;
+  timestamp: Date | null;
+  type: string;
+  user_id: number | null;
+  user_text: string | null;
+}
+
+
+interface SpannerFieldHandler<T> {
```

Comment:
If you make this an abstract class instead of an interface, you can just define the constructor here and not have to repeat it for all the subclasses.